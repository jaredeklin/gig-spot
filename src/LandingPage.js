import React, { Component } from 'react';
import { NavLink, withRouter } from 'react-router-dom';
import { connect } from 'react-redux';
import { setLocation } from './actions';

export class LandingPage extends Component {
  constructor() {
    super()
    this.state ={ 
      zipCode: '',
      radius: ''
    }
  }

  handleChange = (event) => {
    console.log(this.props)
    const { name, value } = event.target;
    this.setState({
      [name]: value
    });
  };

  handleSubmit = (event) => {
    event.preventDefault();
    this.props.setLocation(this.state);
    this.props.history.push('/main');
  }



  render() {
    return (
      <form onSubmit={this.handleSubmit} className='location-form'>
        <input 
          type='text'
          name='zipCode'
          value={this.state.zipCode}
          onChange={this.handleChange}
          placeholder='Zip-code'
        />
        <input
          type='text'
          name='radius'
          value={this.state.radius}
          onChange={this.handleChange}
          placeholder='Radius'
        />
        <button>Submit</button>
      </form>
    )
  }
}

export const mapDispatchToProps = (dispatch) => {
  return {
    setLocation: (location) => (dispatch(setLocation(location)))
  };
};

export default connect(null, mapDispatchToProps)(LandingPage);
