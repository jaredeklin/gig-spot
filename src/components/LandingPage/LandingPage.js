import React from 'react';
import LocationForm from '../../containers/LocationForm/LocationForm';

export const LandingPage = () => (
  <section className="landing-page-background">
    <article className="landing-page">
      <h1>Gig Spot</h1>
      <p>Enter your city find concerts in your area</p>
      <LocationForm />
    </article>
  </section>
);
