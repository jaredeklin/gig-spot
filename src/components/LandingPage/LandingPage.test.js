import React from 'react';
import { shallow } from 'enzyme';
import { LandingPage } from './LandingPage';

describe('LandingPage', () => {
  it('should match the snapshot', () => {
    const wrapper = shallow(<LandingPage />);

    expect(wrapper).toMatchSnapshot();
  });
});
