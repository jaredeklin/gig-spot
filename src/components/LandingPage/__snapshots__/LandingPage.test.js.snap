// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LandingPage should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <article
        className="landing-page"
>
        <h1>
                Gig Spot
        </h1>
        <p>
                Enter your city find concerts in your area
        </p>
        <withRouter(Connect(LocationForm)) />
</article>,
      "className": "landing-page-background",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            Gig Spot
</h1>,
          <p>
            Enter your city find concerts in your area
</p>,
          <withRouter(Connect(LocationForm)) />,
        ],
        "className": "landing-page",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Gig Spot",
          },
          "ref": null,
          "rendered": "Gig Spot",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Enter your city find concerts in your area",
          },
          "ref": null,
          "rendered": "Enter your city find concerts in your area",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "article",
    },
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <article
          className="landing-page"
>
          <h1>
                    Gig Spot
          </h1>
          <p>
                    Enter your city find concerts in your area
          </p>
          <withRouter(Connect(LocationForm)) />
</article>,
        "className": "landing-page-background",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h1>
              Gig Spot
</h1>,
            <p>
              Enter your city find concerts in your area
</p>,
            <withRouter(Connect(LocationForm)) />,
          ],
          "className": "landing-page",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Gig Spot",
            },
            "ref": null,
            "rendered": "Gig Spot",
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Enter your city find concerts in your area",
            },
            "ref": null,
            "rendered": "Enter your city find concerts in your area",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "article",
      },
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
